### PR TITLE
Add toggleReportCellBorders script to input gestures.

### LINKED ISSUE #9507: Input Gestures Dialog: Add Missing Gestures for Document Formatting
When taking a look at the gestures listed for Document Formatting from within the Input Gestures dialog, I've noticed that both "Cell Borders" and "Font Name" are missing. Would it be possible to add these so that all gestures are present?

### PR DESCRIPTION
**Pull request by jakubl7545**
### Link to issue number:
Closes #9507 
### Summary of the issue:
Currently script for reporting cell borders is not present in input gestures dialog so it is not possible to add a gesture for that.
### Description of how this pull request fixes the issue:
It adds toggleReportCellBorders script to globalCommands.
### Testing performed:
I opened document formatting group in input gestures and found added script there. Then assigned gesture and checked that it works.
### Known issues with pull request:
None.
### Change log entry:
It is now possible to assign gesture to script reporting cell borders.
Section: Bug fixes

### CODE REVIEW

**Comment by jakubl7545** [comment]
I closed previous PR and open a new one but it fails with the same message:
Error sending GitHubPullRequest notification: Error commenting on GitHub pull request: {"message":"Validation Failed","errors":[{"resource":"IssueComment","code":"unprocessable","field":"data","message":"Body cannot be blank"}],"documentation_url":"https://developer.github.com/v3/issues/comments/#create-a-comment"}
Any help with that will be appreciated.

File: source/globalCommands.py [hunk 1]
```diff
@@ -567,6 +566,26 @@ def script_toggleReportTableCellCoords(self,gesture):
...
 
+	@script(
+		# Translators: Input help mode message for toggle report cell borders command.
+		description=_("Cycles through cell borders settings"),
+		category=SCRCAT_DOCUMENTFORMATTING,
+	)
+	def script_toggleReportCellBorders(self, gesture):
+		if not config.conf["documentFormatting"]["reportBorderStyle"] and not config.conf["documentFormatting"]["reportBorderColor"]:
+			# Translators: A message reported when cycling through cell borders settings.
+			config.conf["documentFormatting"]["reportBorderStyle"] = True
```
Comment (lukaszgo1) [inline source/globalCommands.py]: At the moment check for translatable commends fails because the commend is not placed directly above the message.

File: source/globalCommands.py [hunk 2]
```diff
@@ -1,9 +1,8 @@
 # -*- coding: UTF-8 -*-
```
Comment (leonardder) [inline source/globalCommands.py]: Feel free to remove this line

File: source/globalCommands.py [hunk 3]
```diff
@@ -1,9 +1,8 @@
 # -*- coding: UTF-8 -*-
-#globalCommands.py
 #A part of NonVisual Desktop Access (NVDA)
```
Comment (leonardder) [inline source/globalCommands.py]: Could you please add a space after every hash sign while at it?

```suggestion
# A part of NonVisual Desktop Access (NVDA)
```

File: source/globalCommands.py [hunk 4]
```diff
@@ -327,7 +326,7 @@ def script_toggleReportFontName(self,gesture):
 			config.conf["documentFormatting"]["reportFontName"]=True
 		ui.message(state)
 	# Translators: Input help mode message for toggle report font name command.
-	script_toggleReportFontName.__doc__=_("Toggles on and off the reporting of font changes")
+	script_toggleReportFontName.__doc__=_("Toggles on and off the reporting of font name changes")
```
Comment (leonardder) [inline source/globalCommands.py]: As you are touching code for this script, could you please also add the script decorator to this one?

File: source/globalCommands.py [hunk 5]
```diff
@@ -567,6 +566,26 @@ def script_toggleReportTableCellCoords(self,gesture):
 	script_toggleReportTableCellCoords.__doc__=_("Toggles on and off the reporting of table cell coordinates")
 	script_toggleReportTableCellCoords.category=SCRCAT_DOCUMENTFORMATTING
 
+	@script(
+		# Translators: Input help mode message for toggle report cell borders command.
+		description=_("Cycles through cell borders settings"),
```
Comment (leonardder) [inline source/globalCommands.py]:
```suggestion
		description=_("Cycles through the cell border reporting settings"),
```

**Review by LeonarddeR — CHANGES_REQUESTED** [review]
Thanks for taking this

File: source/globalCommands.py (hunk 2 above)
Comment (jakubl7545) [reply source/globalCommands.py]: It must be present because there is a letter Ł in one of contributors name.

**Comment by jakubl7545** [comment]
@leonardder All comments are addressed. And thanks a lot @lukaszgo1 for help.

File: source/globalCommands.py (hunk 2 above)
Comment (lukaszgo1) [reply source/globalCommands.py:1]: Now  the encoding declaration can be removed, since the encoding is specified when invoking Gettext.

**Comment by lukaszgo1** [comment]
@jakubl7545 Do you intent to fix the merge conflicts here?

**Comment by lukaszgo1** [comment]
Is there anything blocking this PR from being looked at? cc @seanbudd

File: source/globalCommands.py [hunk 6]
```diff
@@ -702,6 +702,32 @@ def script_toggleReportTableCellCoords(self,gesture):
 			config.conf["documentFormatting"]["reportTableCellCoords"]=True
 		ui.message(state)
 
+	@script(
+		# Translators: Input help mode message for toggle report cell borders command.
+		description=_("Cycles through  the cell border reporting settings"),
```
Comment (seanbudd) [inline source/globalCommands.py:707]:
```suggestion
		description=_("Cycles through the cell border reporting settings"),
```

File: source/globalCommands.py [hunk 7]
```diff
@@ -702,6 +702,32 @@ def script_toggleReportTableCellCoords(self,gesture):
...
 
+	@script(
+		# Translators: Input help mode message for toggle report cell borders command.
+		description=_("Cycles through  the cell border reporting settings"),
+		category=SCRCAT_DOCUMENTFORMATTING,
+	)
+	def script_toggleReportCellBorders(self, gesture):
+		if (
+			not config.conf["documentFormatting"]["reportBorderStyle"] and not
+			config.conf["documentFormatting"]["reportBorderColor"]
```
Comment (seanbudd) [inline source/globalCommands.py:713]: Just to improve readability

```suggestion
			not config.conf["documentFormatting"]["reportBorderStyle"] and
			not config.conf["documentFormatting"]["reportBorderColor"]
```

File: source/globalCommands.py [hunk 8]
```diff
@@ -702,6 +702,32 @@ def script_toggleReportTableCellCoords(self,gesture):
...
+		):
+			# Translators: A message reported when cycling through cell borders settings.
+			ui.message(_("Report styles of cell borders"))
+			config.conf["documentFormatting"]["reportBorderStyle"] = True
+		elif (
+			config.conf["documentFormatting"]["reportBorderStyle"] and not
+			config.conf["documentFormatting"]["reportBorderColor"]
```
Comment (seanbudd) [inline source/globalCommands.py:720]:
```suggestion
			config.conf["documentFormatting"]["reportBorderStyle"] and
			not config.conf["documentFormatting"]["reportBorderColor"]
```

**Review by seanbudd — CHANGES_REQUESTED** [review]
Thanks  @jakubl7545, looks good to me otherwise

**Comment by jakubl7545** [comment]
@seanbudd Changes applied.

File: source/globalCommands.py [hunk 9]
```diff
@@ -382,8 +382,8 @@ def script_toggleSpeakCommandKeys(self,gesture):
 
 	@script(
 		# Translators: Input help mode message for toggle report font name command.
-		description=_("Toggles on and off the reporting of font changes"),
-		category=SCRCAT_DOCUMENTFORMATTING
+		description=_("Toggles on and off the reporting of font name changes"),
+		category=SCRCAT_DOCUMENTFORMATTING,
```
Comment (CyrilleB79) [inline source/globalCommands.py:386]: Why did you change the description of this script? It is unrelated to this PR and it is not described in the PR description.
Moreover, technically, what is reported is really a font change, not the name change of a font that would remain the same but that would just be renamed. Thus this modification doesn't make sense to me.

Thus, I suggest to revert it:

```suggestion
-		description=_("Toggles on and off the reporting of font name changes"),
-		category=SCRCAT_DOCUMENTFORMATTING,
+		description=_("Toggles on and off the reporting of font changes"),
+		category=SCRCAT_DOCUMENTFORMATTING
```

**Comment by jakubl7545** [comment]
That was discussed long time ago. Don't remember the details but I was asked to do that by the reviewer.

File: source/globalCommands.py (hunk 9 above)
Comment (seanbudd) [reply source/globalCommands.py:386]: >That was discussed long time ago. Don't remember the details but I was asked to do that by the reviewer.

@jakubl7545 
I can't see a justification for this on this PR, #9507 or  #10407, so I think removing this is the best course of action. I'm going to commit this suggestion and update the changes so this can get merged.

Comment (seanbudd) [reply source/globalCommands.py]: I see the mention here actually, but I don't think this advice should be followed. 

>Reporting of font names is present, but represented as font changes. I propose this be changed.

**Review by seanbudd — APPROVED** [review]
